app/partials/project-modal-editor.cjsx

```diff
@@ -78,15 +78,33 @@ ProjectModalEditor = createReactClass
       console.log 'ProjectModalEditor onStepOrderChange', arguments
 
   getInitialState: ->
+    error: ''
+    frequency: if @props.projectModal?.configuration?.minicourse_frequency then @props.projectModal?.configuration?.minicourse_frequency.join "," else ''
     stepToEdit: 0
 
   onClick: (stepIndex) ->
     @setState stepToEdit: stepIndex
 
   changeFrequency: (e) ->
-    @props.projectModal.update
-      'configuration.minicourse_frequency': e.target.value
-    @props.projectModal.save()
+    if @state.error then @setState { error: '' }
+    @setState { frequency: e.target.value }
+
+  saveFrequency: (e) ->
+    e.preventDefault()
+    if @state.error then @setState { error: '' }
+    allowedString = /^(\d(,\d)*)$/
+    if allowedString.test(@state.frequency)
+      splitString = @state.frequency.split(',')
+      frequency = splitString.map((digit) => parseInt(digit))
+      @props.projectModal.update
+        'configuration.minicourse_frequency': frequency
+      @props.projectModal.save()
+    else if !@state.frequency
+      @props.projectModal.update
+        'configuration.minicourse_frequency': undefined
```

Comment:
Any reason for `undefined` rather than `null` here? To me, 'null' seems more explicit when setting a value and it's a bit odd to set something to `undefined`.

Reply:
The underlying json-api-client deletes the object property when it's set to undefined:

https://github.com/zooniverse/json-api-client/blob/master/src/model.coffee#L49

which is what I want to do. We do it a few other places:

https://github.com/zooniverse/Panoptes-Front-End/blob/04f4fda0efbc05d0177f6e2c3d38b2b6da22ca09/app/classifier/classifier.jsx#L76

In addition, depending on what you're trying to represent in the state of something, I think `undefined` can be safer than `null` since `null` is still an object.